chart.js.d.ts

```diff
@@ -0,0 +1,605 @@
+
+declare module "chart.js" {
+  export = Chart
+}
+
+declare namespace Chart {
+
+  function Line(context: CanvasRenderingContext2D, args: {data: LineChartData, options?: LineChartOptions}): LineChartInstance;
+
+  function Bar(context: CanvasRenderingContext2D, args: {data: BarChartData, options?: BarChartOptions}): BarChartInstance;
+
+  function Radar(context: CanvasRenderingContext2D, args: {data: RadarChartData, options?: RadarChartOptions}): RadarChartInstance;
+
+  function PolarArea(context: CanvasRenderingContext2D, args: {data: PolarAreaChartData, options?: PolarAreaChartOptions}): PolarAreaChartInstance;
+
+  function Bubble(context: CanvasRenderingContext2D, args: {data: CircularChartData, options?: CircularChartOptions}): CircularChartInstance;
+
+  function Doughnut(context: CanvasRenderingContext2D, args: {data: CircularChartData, options?: CircularChartOptions}): CircularChartInstance;
+
+  const defaults: {
+    global: ChartOptions
+    line: LineChartOptions
+    bar: BarChartOptions
+    radar: RadarChartOptions
+    polarArea: PolarAreaChartOptions
+    doughnut: CircularChartOptions
+    pie: CircularChartOptions
+  };
+
+  // Check required, what fields does the defaultConfig object contain?
+  const scaleService: {
+    registerScaleType<T extends Scale>(key: string, scale: T, defaultConfig: Object): void
+  };
+
+  interface ChartData {
+    labels: Array<string>
+    datasets?: Array<ChartDataSet>
+  }
+
+  interface ChartDataSet {
+    data: Array<number>
+  }
+
+  interface ChartOptions {
+    responsive?: boolean
+    responsiveAnimationDuration?: number
+    maintainAspectRatio?: boolean
+    events?: Array<string>
+    hover?: {
+      onHover?: (activeElements: Array<Element>) => void // Check required, what fields does an element contain? element interface?
+      mode?: string
+      animationDuration?: number
+    }
+    onClick?: (activeElements: Array<Element>) => void // Check required
+    defaultColor?: string
+    defaultFontColor?: string
+    defaultFontFamily?: string
+    defaultFontSize?: number
+    defaultFontStyle?: string
+    legendCallback?: (chart: ChartInstance) => string // Check required
```

Comment:
We provide a default implementation for all chart. Will only be used if `chartInstance.generateLegend()` is called